new PR

RomanKostivAimtraction opened the pull request.

roman120487 committed:
new PR

roman120487 committed:
all rows and selected rows thorught STORE

Lertis commented on src/app/pages/main/main.component.ts:
```diff
@@ -103,9 +103,12 @@ export class MainComponent implements OnInit, OnDestroy {
+    const countRow: number = this.gridOptions.api.getSelectedRows().length;
     // this.countSelectedCase = this.gridOptions.api.getSelectedRows().length;
-    this.store.dispatch(countSelectedRows());
+    console.log('countRow');
```
console.log should be removed before merge

roman120487 committed:
add 404 page

roman120487 committed:
reactive Form ensted of ngModel

roman120487 committed:
style width angular material and bootstrap

roman120487 committed:
add style grid, fix change mode btn

roman120487 committed:
complate some remarks

- add wildcard check (completed)
- replace [(ngModel)] usage with reactive forms (completed)
- "all rows" - field is editable. For a what reason? The same for selected rows (completed)
- Restrict ability to search with empty input search field (completed)
- After changing a mode - selected rows are still selected (Bug, I guess) (completed)
- use different approach with for providedIn: 'root'(completed)

roman120487 committed:
fix mistakes with lint

Lertis commented on src/app/pages/main/main.component.ts:
```diff
@@ -62,19 +56,24 @@ export class MainComponent implements OnInit, OnDestroy {
+    private domSanitizer: DomSanitizer, // add custom icon
+    private toastr: ToastrService
   ) {
+    this.matIconRegistry.addSvgIcon('search', this.domSanitizer.bypassSecurityTrustResourceUrl('../assets/icons/search.svg'));
```
Good job!)

roman120487 committed:
update many thisngs ))

completed http req throuth store,
make spiner width store,
update servise,
optimised main component,
update effect store

roman120487 committed:
change location store

roman120487 committed:
try create test for store and reactive form

roman120487 committed:
covered unitest for main.app

roman120487 committed:
update test
